### PR TITLE
PLT - Add objective description

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
This adds the possibility to provide details about the objective metrics, for instance, value and support size.

It exposes an icon beside ``objective_column`` which when hovered shows a description of the objective metrics.
[Live preview of the feature](https://badr-moufad.github.io/share-benchmarks/prev_objective_desc/generated_df_lasso_bench.html).

------
revival of #523

### CODE REVIEW

**Comment by Badr-MOUFAD** [comment]
Referring to #523, it was suggested to place the description at the bottom of the sidebar.

WDYT about this way of tackling that? I believe that the sidebar is more valuable for adding visualization capabilities than showing details.

**Comment by mathurinm** [comment]
I like the current design! Would you display information about all objectives possible, or only the selected one?

**Comment by Badr-MOUFAD** [comment]
Here is [a preview of the new feature](https://badr-moufad.github.io/share-benchmarks/prev_objective_desc/dummy_bench_benchopt_run_2023-05-22_13h33m23.html).
Also, I ensured it won't break the older versions.

**Review by mathurinm — APPROVED** [review]

**Comment by mathurinm** [comment]
Following discussion with @tomMoral can you try an option where the description is more visible, ie:
below the selector box, showing the first line of the description, and hovering would expand the box?

**Comment by tomMoral** [comment]
maybe not expanding on hovering but showing the tool tip?
I just find it hard to discover that hovering on the question mark shows the description?

**Comment by Badr-MOUFAD** [comment]
My apologies @tomMoral, I'm not sure if I understand well what you mean.

**Comment by tomMoral** [comment]
It took me a while to understand where the description where because the `?` button is very small and not super intuitive IMO.
image

I would rather have the first line of the docstring displayed bellow the selectbox, and when hovering on this line, the current popup that displays the full description.
WDYT?

**Comment by mathurinm** [comment]
I'm +1 on @Badr-MOUFAD's original suggestion here (the plain question mark without effect nor "hover over"), the question mark is explicit enough in my opinion

**Comment by tomMoral** [comment]
Sorry for the late reply.
I understand the argument that it might clug the display if we do this for every drop down menu. For the current suggestion:
- I don't like the pulse effect, it takes too much attention while we just want the feature to be discoverable.
- Instead of `(hover over)`, It would simply put `description`, it is explicit enough?

**Comment by mathurinm** [comment]
@Badr-MOUFAD nice! 
can you add a line in the whatsnew and a small example documentation ?

**Comment by mathurinm** [comment]
Just tested it: 
- the question mark button needs clicking. Hovering seems more natural, WDYT? Don't know if people use it on mobile though
- can you configure the box size so that it has a length of 80 characters? in its current state it's a bit short, while we allow the docstring to go up to 80 characters: 
image

**Comment by Badr-MOUFAD** [comment]
> the question mark button needs clicking. Hovering seems more natural, WDYT? Don't know if people use it on mobile though

Clicking would work for mobile. Do you suggest switching completely to click instead of hover?

> can you configure the box size so that it has a length of 80 characters? in its current state it's a bit short, while we allow the docstring to go up to 80 characters:

Yes indeed, it makes sense to set 80 characters long for the description box

**Comment by tomMoral** [comment]
Do you understand why there is no description in this example:
https://output.circle-artifacts.com/output/job/0f815dbb-0e04-4aee-8d97-1107c2f3afcf/artifacts/0/result.html

should you add a description in the example result to see it?

**Comment by tomMoral** [comment]
the exposed artifact is generated with this call:
`benchopt run benchopt/tests/test_benchmarks/dummy_benchmark -d Simulated[n_features=100,n_samples=100,rho=0]  -o "*[reg=0.1]" -s python-pgd -s cd -s Solver-Test -r 5 -n 15 --output benchmark-result`

**Comment by Badr-MOUFAD** [comment]
> Do you understand why there is no description in this example: https://output.circle-artifacts.com/output/job/0f815dbb-0e04-4aee-8d97-1107c2f3afcf/artifacts/0/result.html
> 
> should you add a description in the example result to see it?


Nothing will be shown if the description was not provided. [code ref](https://github.com/Badr-MOUFAD/benchopt/blob/c01e80c7025a379d6171d7e5a3635adfdb74bc2b/benchopt/plotting/html/templates/result.mako.html#L61-L69)

Do you prefer to show it anyways, for instance "Description not provided"?

**Comment by tomMoral** [comment]
No it is fine like that but can you update the example in the dummy dataset so it shows and we can check the feature o. the webpage?

File: benchopt/plotting/html/templates/result.mako.html [hunk 1]
```diff
@@ -55,7 +55,19 @@ <h1 class="ml11">
                             </div>
                         </div>
                         <div class="mt-8 px-8">
-                            <label for="objective_column" class="block text-sm font-medium text-white">Objective column</label>
+                            <div class="flex flex-row items-center justify-between text-sm font-medium text-white">
+                                <label for="objective_column" class="block">Objective column</label>
+                                <!--don't show description unless it is provided -->
+                                % if result['metadata']['obj_description'] != "":
+                                    <span class="objective-description-trigger" style="cursor: pointer;">
+                                        <span style="margin-right: 0.5em;">(description)</span>
```
Comment (tomMoral) [inline benchopt/plotting/html/templates/result.mako.html:63]: nitpick: remove the parenthesis:
```suggestion
                                        <span style="margin-right: 0.5em;">description</span>
```

File: benchopt/runner.py [hunk 2]
```diff
@@ -187,6 +187,13 @@ def run_one_solver(benchmark, dataset, objective, solver, n_repetitions,
     if stopping_strategy == 'callback':
         stopping_strategy = 'iteration'
 
+    # get objective description
+    # use `obj_` instead of `objective_` to avoid conflicts with
+    # the name of metrics in Objective.compute
+    obj_description = objective.__doc__
+    if obj_description is None:
+        obj_description = ""
```
Comment (tomMoral) [inline benchopt/runner.py:195]:
```suggestion
    obj_description = objective.__doc__ or ""
```

**Review by tomMoral — APPROVED** [review]
A few nitpicks but otherwise LGTM

**Comment by tomMoral** [comment]
Thanks a lot @Badr-MOUFAD  :)

**Pull request merged**